lib/src/utils/screen_loading/screen_loading_manager.dart

```diff
@@ -183,10 +200,7 @@ class ScreenLoadingManager {
         return;
       }
 
-      final isSameScreen = RouteMatcher.I.match(
-        routePath: trace.screenName,
-        actualPath: currentUiTrace?.screenName,
-      );
+      final isSameScreen = currentUiTrace?.matches(trace.screenName) == true;
```

Comment:
Replaced the direct `RouteMatcher` call with `UiTrace.matches` as the matching screen name is not exposed on the `UiTrace` object to avoid misuse.